### PR TITLE
Added OpFlashes and XARAPuCA flashes to SimpleFlash

### PR DESCRIPTION
**Pull request by Robert-Darby**
- Added OpFlashes to SimpleFlash for SBND and ICARUS
- Added XARAPUCA flashes to SimpleFlash for SBND
- Updated SRSlice to accoutn for new flashes in CAF

### CODE REVIEW

**Comment by Robert-Darby** [comment]
Related PRs:
icaruscode: https://github.com/SBNSoftware/icaruscode/pull/621
sbndcode: https://github.com/SBNSoftware/sbndcode/pull/381
sbncode: https://github.com/SBNSoftware/sbncode/pull/382
sbnanaobj: https://github.com/SBNSoftware/sbnanaobj/pull/107

File: sbndcode/FlashMatch/flashmatch_sbnd.fcl [hunk 1]
```diff
@@ -114,4 +119,62 @@ sbnd_simple_flashmatch_sce.TrackProducer: "pandoraSCETrack"
...
+sbnd_simple_flashmatch_ara.BeamSpillTimeEnd: 1.8
+
+sbnd_simple_flashmatch_ara_sce: @local::sbnd_simple_flashmatch
+sbnd_simple_flashmatch_ara_sce.FlashType: "simpleflash_ara"
+sbnd_simple_flashmatch__arasce.PandoraProducer: "pandoraSCE"
+sbnd_simple_flashmatch_ara_sce.TrackProducer: "pandoraSCETrack"
+sbnd_simple_flashmatch_ara_sce.CaloProducer: "pandoraSCECalo"
+sbnd_simple_flashmatch_ara_sce.SpacePointProducer: "pandoraSCE"
+sbnd_simple_flashmatch_ara_ace.OpHitProducer: "ophitxarapuca"
+sbnd_simple_flashmatch_ara_sce.OpFlashProducer: ["opflashtpc0xarapuca", "opflashtpc1xarapuca"]
```
Comment (ikatza) [inline sbndcode/FlashMatch/flashmatch_sbnd.fcl:137]: Typos here.

File: sbndcode/FlashMatch/flashmatch_sbnd.fcl [hunk 2]
```diff
@@ -114,4 +119,62 @@ sbnd_simple_flashmatch_sce.TrackProducer: "pandoraSCETrack"
...
+sbnd_simple_flashmatch_opara.OpHitProducer: "ophitxarapuca"
+sbnd_simple_flashmatch_opara.OpFlashProducer: ["opflashtpc0xarapuca", "opflashtpc1xarapuca"]
+sbnd_simple_flashmatch_opara.OpFlashHitProducer: ["opflashtpc0xarapuca", "opflashtpc1xarapuca"]
+sbnd_simple_flashmatch_opara.BeamSpillTimeStart: 0.
+sbnd_simple_flashmatch_opara.BeamSpillTimeEnd: 1.65
+
+sbnd_simple_flashmatch_opara_sce: @local::sbnd_simple_flashmatch
+sbnd_simple_flashmatch_opara_sce.IsSimple: false
+sbnd_simple_flashmatch_opara_sce.FlashType: "opflash_ara"
+sbnd_simple_flashmatch_sce_sce.PandoraProducer: "pandoraSCE"
```
Comment (ikatza) [inline sbndcode/FlashMatch/flashmatch_sbnd.fcl:170]: Typo?

File: sbndcode/JobConfigurations/base/reco_sbnd.fcl [hunk 3]
```diff
@@ -176,6 +176,12 @@ physics:
   ### flash-matching
   fmatch:              @local::sbnd_simple_flashmatch
   fmatchSCE:           @local::sbnd_simple_flashmatch_sce
+  fmatchop:              @local::sbnd_simple_flashmatch_op
+  fmatchopSCE:           @local::sbnd_simple_flashmatch_op_sce
+  fmatchara:              @local::sbnd_simple_flashmatch_ara
+  fmatcharaSCE:           @local::sbnd_simple_flashmatch_ara_sce
+  fmatchopara:              @local::sbnd_simple_flashmatch_opara
+  fmatchoparaSCE:           @local::sbnd_simple_flashmatch_opara_sce
```
Comment (ikatza) [inline sbndcode/JobConfigurations/base/reco_sbnd.fcl:184]: Review indentation of second column

**Review by ikatza — CHANGES_REQUESTED** [review]
Typos

**Comment by Robert-Darby** [comment]
New metrics file for SBND: /sbnd/data/users/rdarby/flashmatch/opflash/nonlin/all_flash/FlashMatch/

**Comment by fjnicolas** [comment]
Thanks @Robert-Darby, I will work on the `sbnd_data` update tomorrow

**Comment by ikatza** [comment]
Looking at those metrics there's an oddity with the ratio metric for PMTs. The ratio should be equal or less than 1, but looking at the TH2 for simple flash and opflash it's clear that the ratio can take values higher than 1.

For now I suggest to postpone the investigation of the oddity, but update the upper limit of the histogram so that all data points get considered. In `flashmatch_sbnd.fcl` change `ratio.up` to 1.5 (maybe 2?)

New metrics need to be created.

**Comment by ikatza** [comment]
> Thanks @Robert-Darby, I will work on the `sbnd_data` update tomorrow

It's still too early for that.

**Comment by fjnicolas** [comment]
@ikatza the pmt detection efficiencies were updated some months ago. The efficiency for each flavour of PMT is now different (we were using a global efficiency of 3% for both coated/uncoated), meaning that the ratio metric is no longer constrained to [0, 1]. The flash matching metrics were updated accordingly in `sbnd_data`.

```
  # Detection efficiencies
  PMTCoatedVUVEff:       0.12       #PMT coated detection efficiency for direct (VUV) light
  PMTCoatedVISEff:       0.13       #PMT coated detection efficiency for reflected (VIS) light
  PMTUncoatedEff:        0.19       #PMT uncoated detection efficiency
```

@Robert-Darby @ikatza let me know when we are ready to update the metric files.

**Review by ikatza — APPROVED** [review]

**Comment by fjnicolas** [comment]
trigger build SBNSoftware/sbndcode#395 SBNSoftware/sbncode#382 SBNSoftware/sbnanaobj#107

**Comment by fjnicolas** [comment]
Hi @Robert-Darby, I see in the CI that the `fmatch` is not filled as it used to be, e.g.,
```
338: Checking record 0
339:   rec.slc[0].fmatch.chargeCenter.x differs: 45.48662567138672 vs nan
340:   rec.slc[0].fmatch.chargeCenter.y differs: -2.4763777256011963 vs nan
341:   rec.slc[0].fmatch.chargeCenter.z differs: 258.91595458984375 vs nan
342:   rec.slc[0].fmatch.chargeQ differs: 1765561.5 vs nan
343:   rec.slc[0].fmatch.chargeWidth.x differs: 1.3069051504135132 vs nan
344:   rec.slc[0].fmatch.chargeWidth.y differs: 1.7959880828857422 vs nan
345:   rec.slc[0].fmatch.chargeWidth.z differs: 147.6423797607422 vs nan
```
is that expected?

**Comment by fjnicolas** [comment]
trigger build SBNSoftware/sbncode#382 SBNSoftware/sbnanaobj#107

**Pull request merged**